modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -674,12 +693,41 @@ private int getUploadIndex(String targetUploadId, List<MultipartUpload> multipar
             return found ? uploadIndex : -1;
         }
 
-        void run(BytesReference expected, BytesReference updated, ActionListener<OptionalBytesReference> listener) throws Exception {
+        /**
+         * @return {@code true} if there are already ongoing uploads, so we should not proceed with the operation
+         */
+        private boolean checkPreexistingUploads() {
+            final var uploads = listMultipartUploads();
+            if (uploads.isEmpty()) {
+                return false;
+            }
+
+            final var expiryDate = Date.from(
+                Instant.ofEpochMilli(
+                    blobStore.getThreadPool().absoluteTimeInMillis() - blobStore.getCompareAndExchangeTimeToLive().millis()
```

Comment:
I guess we can tolerate a clock drift between the S3 server and the local node clock in this case? the default expiry time is relatively high.

Reply:
Aha, I've just seen the log message that warns about this potential issue